src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -1010,10 +1010,12 @@ private static void NarrowFourUtf16CharsToAsciiAndWriteToBuffer(ref byte outputB
         {
             Debug.Assert(AllCharsInUInt64AreAscii(value));
 
-            if (Bmi2.X64.IsSupported)
+            if (Sse2.X64.IsSupported)
             {
-                // BMI2 will work regardless of the processor's endianness.
-                Unsafe.WriteUnaligned(ref outputBuffer, (uint)Bmi2.X64.ParallelBitExtract(value, 0x00FF00FF_00FF00FFul));
+                Debug.Assert(BitConverter.IsLittleEndian, "SSE2 narrowing assumes little-endian.");
```

Comment:
I don't think this is necessary. It is highly unlikely that an x86 computer would ever start supporting big endian and we have other bugs if that did ever come to pass.